### ISSUE TITLE
Log messages do not show source(file:line)

### ISSUE THREAD

**Issue by PonderingGrower** [pimterry/loglevel#96]
Hi,

I'm really liking the simplicity of your library but it looks to me that when I use it the Firefox console(haven't tested other browsers) does not show the source of the message - meaning file and line from which it came. It shows up only as `(unknown)`.

Screenshot:
![loglevel_unknown](https://cloud.githubusercontent.com/assets/2212681/17402175/aebbcb8e-5a49-11e6-9187-20daf0ba43f0.png)

Is this fixable? Is there anything I can do in my code to make it work?

Cheers!

**Comment by pimterry** [comment]
Can you include the code you're using to do this? And the version number of your firefox?

I'm not sure why this is happening. I've done some quick tests and it looks like it's a recent change in Firefox. In version 44, all log messages come through fine with lines, but in version 48 they don't.

It seems like maybe firefox has just disabled tracing entirely. `console.warn("hi")` for me in Firefox 48 doesn't include any lines at all, and `console.warn.bind(console)("hi")` - more or less the same as what loglevel is doing internally - prints 'unknown' in the same way.

Long story short, it looks like a firefox problem. I'm happy to implement workarounds if you can think of them, and find a good way to detect this case. You do still get stack traces from `console.error`, and thereby `log.error` too, which could be a potential solution for your code for now. The right solution is probably to talk to the firefox team though, and get them to put log source tracing back in.

**Comment by PonderingGrower** [comment]
Sorry for the late reply but I was out.

I'll check if this is Firefox specific and if bound to a certain version tomorrow at work. Cheers!

**Comment by PonderingGrower** [comment]
So I just tested this on my linux desktop at work which runs `48.0` and I get `(unknown)`. Also checked with a friend that was running `47.0` on a Mac and it did the same thing.

But when I test it on my laptop which run `45.3.0` I see source of the message just fine.

![loglevel_ff_45 3](https://cloud.githubusercontent.com/assets/2212681/17510641/bb8cd0c0-5e17-11e6-962e-5b3cda70d03d.png)

Seems like something that Firefox broke with one of it's recent releases.

The way I use your code is kinda convoluted. I build a GreaseMonkey script using [webpack](https://webpack.github.io/) out of multiple modules through the CommonJS pattern and I include your module through a `LOG = require('loglevel')` call which results in a code that has multiple layers of modules using function closure.

Aside from that just run `LOG.info('whatever')` like any normal person would.

**Comment by ftaiolivista** [comment]
same problem on Google Chrome 54.0.2840.100 64BIT on ubuntu
